### PR TITLE
Implement more Numpy functions

### PR DESCRIPTION
**Pull request by pitrou**
This PR adds support for the Numpy functions diff(), digitize(), searchsorted(), bincount(), histogram(), nanmin(), nanmax(), nansum(), nanmean(), nanstd(), nanvar(), nanmedian().

Based on PR #1789.

### CODE REVIEW

File: numba/targets/arraymath.py [hunk 1]
```diff
@@ -345,14 +286,261 @@ def flat_all(a):
...
         return False
 
     return flat_any
 
 
+def get_isnan(dtype):
+    """
+    A generic isnan() function
+    """
+    if isinstance(dtype, types.Number):
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: Perhaps is not Integer ?

File: numba/targets/arraymath.py [hunk 2]
```diff
@@ -345,14 +286,261 @@ def flat_all(a):
...
+
+    def nanmin_impl(a):
+        if a.size == 0:
+            raise ValueError("nanmin(): empty array")
+        for view in numpy.nditer(a):
+            minval = view.item()
+            break
+        for view in numpy.nditer(a):
+            v = view.item()
+            if not minval < v and not isnan(v):
```
Comment (stuartarchibald) [inline numba/targets/arraymath.py]: for comparison operations with nan, they ought return false so the branch ought to execute anyway without the second guard condition?

Comment (pitrou) [inline numba/targets/arraymath.py]: Note the `not` in the first condition.

File: numba/targets/arraymath.py (hunk 1 above)
Comment (pitrou) [inline numba/targets/arraymath.py]: Good catch, fixed.

**Pull request merged**